### PR TITLE
fix show new incoming messages after clearing chat

### LINKED ISSUE #3991: message view not always updated
the message view is not always updated; at least not after "clear" in a reproducible way, not sure about other situations.

to reproduce:

- chose "Clear Chat" from the three-dot-menu in the upper right corner
- send a message to that chat from another device (without doing anything on the cleared chat)

-> the message appears in the chatlist, but the chat stays empty.

dc 1.46.1 on mac

https://github.com/deltachat/deltachat-desktop/assets/9800740/908bf7bd-ac1d-48a4-b78d-00f8f99f3af2

### PR DESCRIPTION
**Pull request by Simon-Laux**
fixes #3991
closes #3994

### CODE REVIEW

File: src/renderer/stores/messagelist.ts [hunk 1]
```diff
@@ -740,20 +740,23 @@ class MessageListStore extends Store<MessageListState> {
...
-        } else {
-          if (item.kind === 'message') {
-            return item.msg_id === (last_item as any).msg_id
-          } else {
-            return item.timestamp === (last_item as any).timestamp
-          }
-        }
-      })
+      let indexStart =
+        last_item === undefined
```
Comment (farooqkz) [inline src/renderer/stores/messagelist.ts:747]: I am in favor of `lastItem` instead of `last_item`.

**Review by farooqkz — CHANGES_REQUESTED** [review]

File: src/renderer/stores/chat/chat_scheduler.ts [hunk 2]
```diff
@@ -119,7 +119,7 @@ export class ChatStoreScheduler {
       try {
         returnValue = await effect(...args)
       } catch (err) {
-        log.error(`Error in queuedEffect ${effectName}: ${err}`)
+        log.error(`Error in queuedEffect ${effectName}:`, err)
```
Comment (farooqkz) [inline src/renderer/stores/chat/chat_scheduler.ts:122]: Who was blaming me for unrelated changes? :D

Comment (Simon-Laux) [reply src/renderer/stores/chat/chat_scheduler.ts:122]: it is related, I needed this to get the real error. basically I reproduced it, got `Error in queuedEffect ...` but only he text of the error without stack trace, this change pointed me to the location in the message list store file.

File: src/renderer/stores/messagelist.ts (hunk 1 above)
Comment (Simon-Laux) [reply src/renderer/stores/messagelist.ts:747]: not really topic of this pr, this pr should just fix the bug, the name is old.

Comment (Simon-Laux) [reply src/renderer/stores/messagelist.ts:747]: we should do the rename in a dedicated pr, otherwise it's hard to see the relevant changes,

**Review by WofWca — APPROVED** [review]
So it's just about `last_item` being possibly `undefined`.

This fixes the issue I described in https://github.com/deltachat/deltachat-desktop/issues/3991 I don't know about the original one.

**Comment by Simon-Laux** [comment]
> This fixes the issue I described in https://github.com/deltachat/deltachat-desktop/issues/3991 I don't know about the original one.

the initial issue by @r10s is: other account sends you messages. his description of "other device" is a bit misleading because you can misunderstand it as multi device (self sent message from same account on other device).

**Comment by WofWca** [comment]
You're right.

**Pull request merged**